pkg/webhooks/customresourcedefinitions/customresourcedefinitions.go

```diff
@@ -0,0 +1,202 @@
+package customresourcedefinitions
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName                    string = "customresourcedefinitions-validation"
+	docString                      string = `Managed OpenShift Customers may not change CustomResourceDefinitions managed by Red Hat.`
+	privilegedServiceAccountGroups string = `^system:serviceaccounts:(kube.*|openshift.*|default|redhat.*|osde2e-[a-z0-9]{5})`
+)
+
+var (
+	timeout                          int32 = 2
+	allowedUsers                           = []string{"system:admin", "backplane-cluster-admin"}
+	sreAdminGroups                         = []string{"system:serviceaccounts:openshift-backplane-srep"}
+	privilegedServiceAccountGroupsRe       = regexp.MustCompile(privilegedServiceAccountGroups)
+	scope                                  = admissionregv1.ClusterScope
+	rules                                  = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+				admissionregv1.Update,
+				admissionregv1.Delete,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"apiextensions.k8s.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"customresourcedefinitions"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log = logf.Log.WithName(WebhookName)
+)
+
+// customresourcedefinitionsruleWebhook validates a customresourcedefinition change
+type customresourcedefinitionsruleWebhook struct {
+	s runtime.Scheme
+}
+
+// We just need a runtime object to get the labels
+type customResourceDefinition struct {
+	runtime.Object
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *customresourcedefinitionsruleWebhook {
+	scheme := runtime.NewScheme()
+	return &customresourcedefinitionsruleWebhook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *customresourcedefinitionsruleWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	//Implement authorized next
+	return s.authorized(request)
+}
+
+func (s *customresourcedefinitionsruleWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	np, err := s.renderCustomResourceDefinition(request)
```

Comment:
```suggestion
	crd, err := s.renderCustomResourceDefinition(request)
```
nit, but the `np` variable name doesn't really make sense to me